file.go

```diff
@@ -0,0 +1,144 @@
+package scribe
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+	"sync"
+	"time"
+
+	"github.com/Hatch1fy/cron"
+	"github.com/Hatch1fy/errors"
+)
+
+const (
+	// ErrFileUnset is returned when a file is not set and attempting to be written to
+	ErrFileUnset = errors.Error("cannot write, file is currently unset")
+	// ErrRotationIntervalSet is returned when a rotation interval is attempting to be set when it's already been set
+	ErrRotationIntervalSet = errors.Error("error setting rotation interval, value already set")
+)
+
+// NewFile will return a new Scribe
+func NewFile(name, dir string) (fp *File, err error) {
+	var f File
+	f.name = name
+	f.dir = dir
+
+	if err = f.openFile(); err != nil {
+		return
+	}
+
+	fp = &f
+	return
```

Comment:
What does this do? Does go implicitly return `fp` since it is named as a return type?

Reply:
This is correct! We utilize named return values when possible here. It helps to keep the return values concise. Sometimes the unnamed returns feel ambiguous.